### PR TITLE
Added Git CLI implementation

### PR DESCRIPTION
**Pull request by avasquez614**

### CODE REVIEW

File: src/main/resources/crafter/studio/studio-config.yaml [hunk 1]
```diff
@@ -87,6 +87,12 @@ studio.repo.ignoreFiles:
 studio.repo.retryingOperation.maxAttempts: 50
 # Max sleep time in milliseconds between retries for retrying repository operation
 studio.repo.retryingOperation.maxSleep: 1000
+# If Git CLI should be used for some Git operations, for better performance (Git CLI needs to be installed beforehand)
+studio.repo.git.cli.enabled: false
```
Comment (sumerjabri) [inline src/main/resources/crafter/studio/studio-config.yaml]: Please default this to true.

File: src/main/java/org/craftercms/studio/api/v2/annotation/RetryingDatabaseOperationAnnotationHandler.java [hunk 2]
```diff
@@ -58,6 +58,8 @@ public void setMaxSleep(int maxSleep) {
 
     @Around("@within(org.craftercms.studio.api.v2.annotation.RetryingDatabaseOperation) || " +
             "@annotation(org.craftercms.studio.api.v2.annotation.RetryingDatabaseOperation)")
+    // TODO: AV - This has the same problem as the old RetryingRepositoryOperationAnnotationHandler. We can just copy
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/api/v2/annotation/RetryingDatabaseOperationAnnotationHandler.java:61]: What's the problem and why leave it in?

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/resources/crafter/studio/studio-config.yaml (hunk 1 above)
Comment (avasquez614) [reply src/main/resources/crafter/studio/studio-config.yaml]: Done

File: src/main/java/org/craftercms/studio/api/v2/annotation/RetryingDatabaseOperationAnnotationHandler.java (hunk 2 above)
Comment (sumerjabri) [reply src/main/java/org/craftercms/studio/api/v2/annotation/RetryingDatabaseOperationAnnotationHandler.java:61]: @jdrossl this is an improvement opportunity to unify the retrying mechanism and make it better.

**Review by sumerjabri — APPROVED** [review]

**Review by jdrossl — APPROVED** [review]

**Pull request merged**